electroncash/rpa/paycode.py

```diff
@@ -0,0 +1,448 @@
+        final_outputs.append((TYPE_ADDRESS, address, amount))
+
+    coins = wallet.get_spendable_coins(domain, config)
+    try:
+        tx = wallet.make_unsigned_transaction(
+            coins, final_outputs, config, fee, change_addr)
+    except BaseException:
+        return 0
+    if locktime is not None:
+        tx.locktime = locktime
+    if not unsigned:
+        run_hook('sign_tx', wallet, tx)
+        wallet.sign_transaction(tx, password)
+    return tx
+
+
+def _calculate_paycode_shared_secret(private_key, public_key, outpoint):
+    """private key is expected to be an integer.
+    public_key is expected to be bytes.
+    outpoint is expected to be a string.
+    returns the paycode shared secret as bytes"""
+
+    from ..bitcoin import Point
+    from ..bitcoin import curve_secp256k1 as curve
+
+    # Public key is expected to be compressed.  Change into a point object.
+    pubkey_point = bitcoin.ser_to_point(public_key)
+    ecdsa_point = Point(curve, pubkey_point.x(), pubkey_point.y())
+
+    # Multiply the public and private points together
+    ecdh_product = ecdsa_point * private_key
+    ecdh_x = int(ecdh_product.x())
+    ecdh_x_bytes = ecdh_x.to_bytes(33, byteorder="big")
+
+    # Get the hash of the product
+    sha_ecdh_x_bytes = sha256(ecdh_x_bytes)
+    sha_ecdh_x_as_int = int.from_bytes(sha_ecdh_x_bytes, byteorder="big")
+
+    # Hash the outpoint string
+    hash_of_outpoint = sha256(outpoint)
+    hash_of_outpoint_as_int = int.from_bytes(hash_of_outpoint, byteorder="big")
+
+    # Sum the ECDH hash and the outpoint Hash
+    grand_sum = sha_ecdh_x_as_int + hash_of_outpoint_as_int
+
+    # Hash the final result
+    nbytes = (len("%x" % grand_sum) + 1) // 2
+    grand_sum_bytes = grand_sum.to_bytes(nbytes, byteorder="big")
+    shared_secret = sha256(grand_sum_bytes)
+
+    return shared_secret
+
+
+def _generate_address_from_pubkey_and_secret(parent_pubkey, secret):
+    """parent_pubkey and secret are expected to be bytes
+    This function generates a receiving address based on CKD."""
+
+    new_pubkey = bitcoin.CKD_pub(parent_pubkey, secret, 0)[0]
+    use_uncompressed = True
+
+    # Currently, just uses compressed keys, but if this ever changes to
+    # require uncompressed points:
+    if use_uncompressed:
+        pubkey_point = bitcoin.ser_to_point(new_pubkey)
+        x_coord = hex(pubkey_point.x())[2:].zfill(64)
+        y_coord = hex(pubkey_point.y())[2:].zfill(64)
+        uncompressed = "04" + \
+            hex(pubkey_point.x())[2:].zfill(64) + \
+            hex(pubkey_point.y())[2:].zfill(64)
+        new_pubkey = bytes.fromhex(uncompressed)
+    return Address.from_pubkey(new_pubkey)
+
+
+def _generate_privkey_from_secret(parent_privkey, secret):
+    """parent_privkey and secret are expected to be bytes
+    This function generates a receiving address based on CKD."""
+
+    return bitcoin.CKD_priv(parent_privkey, secret, 0)[0].hex()
+
+
+def get_grind_string(wallet, prefix_size="10"):
```

Comment:
I don't think there's a setting.  It's just set at "10" (16 bits) in the client code.  It can be modified at the code level, but there is no UI to change this currently and that's good.